### ISSUE TITLE
LicenseDownloadException: Failed to download license

### ISSUE THREAD

**Issue by trampster** [sensslen/nuget-license#16]
I'm getting LicenseDownloadExceptions on two of my projects, one is a Asp.net webapi project and one is a .net maui project.

**Maui Project**
Command `nuget-license -i Pttox.Forms.sln -t -d licenses -o jsonPretty`

Output:
```
Unhandled exception. NuGetUtility.LicenseValidator.LicenseDownloadException: Failed to download license for package System.Buffers (4.3.0).
Context: /home/daniel/Work/hydra-src/apps/Xamarin-PTToX/src/Pttox.Forms/Pttox.Forms.csproj
 ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 429 (Too Many Requests).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at System.Net.Http.HttpClient.GetStreamAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at NuGetUtility.Wrapper.HttpClientWrapper.FileDownloader.DownloadFile(Uri url, String fileName) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/Wrapper/HttpClientWrapper/FileDownloader.cs:line 17
   at NuGetUtility.Wrapper.HttpClientWrapper.FileDownloader.DownloadFile(Uri url, String fileName)
   at NuGetUtility.LicenseValidator.LicenseValidator.ValidateLicenseByUrl(IPackageMetadata info, String context, ConcurrentDictionary`2 result) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 164
   --- End of inner exception stack trace ---
   at NuGetUtility.LicenseValidator.LicenseValidator.ValidateLicenseByUrl(IPackageMetadata info, String context, ConcurrentDictionary`2 result) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 169
   at NuGetUtility.LicenseValidator.LicenseValidator.Validate(IAsyncEnumerable`1 packages) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 47
   at NuGetUtility.LicenseValidator.LicenseValidator.Validate(IAsyncEnumerable`1 packages) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 33
   at NuGetUtility.Program.OnExecuteAsync(CancellationToken cancellationToken) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/Program.cs:line 126
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.InvokeAsync(MethodInfo method, Object instance, Object[] arguments)
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.OnExecute(ConventionContext context, CancellationToken cancellationToken)
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.<>c__DisplayClass0_0.<<Apply>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync(String[] args, CancellationToken cancellationToken)
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync[TApp](CommandLineContext context, CancellationToken cancellationToken)
   at NuGetUtility.Program.Main(String[] args) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/Program.cs:line 86
   at NuGetUtility.Program.<Main>(String[] args)
Aborted (core dumped)
```

**Asp.net webapi project**
Command : `nuget-license -i ApiServer.sln -t -d licenses -o jsonPretty`

Output:
```
Unhandled exception. NuGetUtility.LicenseValidator.LicenseDownloadException: Failed to download license for package NETStandard.Library (2.0.0).
Context: /home/daniel/Work/hydra-src/apps/API-Server/ApiServer.Tests/ApiServer.Tests.csproj
 ---> System.Net.Http.HttpRequestException: Response status code does not indicate success: 404 (Not Found).
   at System.Net.Http.HttpResponseMessage.EnsureSuccessStatusCode()
   at System.Net.Http.HttpClient.GetStreamAsyncCore(HttpRequestMessage request, CancellationToken cancellationToken)
   at NuGetUtility.Wrapper.HttpClientWrapper.FileDownloader.DownloadFile(Uri url, String fileName) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/Wrapper/HttpClientWrapper/FileDownloader.cs:line 17
   at NuGetUtility.Wrapper.HttpClientWrapper.FileDownloader.DownloadFile(Uri url, String fileName) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/Wrapper/HttpClientWrapper/FileDownloader.cs:line 19
   at NuGetUtility.LicenseValidator.LicenseValidator.ValidateLicenseByUrl(IPackageMetadata info, String context, ConcurrentDictionary`2 result) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 164
   --- End of inner exception stack trace ---
   at NuGetUtility.LicenseValidator.LicenseValidator.ValidateLicenseByUrl(IPackageMetadata info, String context, ConcurrentDictionary`2 result) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 169
   at NuGetUtility.LicenseValidator.LicenseValidator.Validate(IAsyncEnumerable`1 packages) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 47
   at NuGetUtility.LicenseValidator.LicenseValidator.Validate(IAsyncEnumerable`1 packages) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/LicenseValidator/LicenseValidator.cs:line 33
   at NuGetUtility.Program.OnExecuteAsync(CancellationToken cancellationToken) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/Program.cs:line 126
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.InvokeAsync(MethodInfo method, Object instance, Object[] arguments)
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.OnExecute(ConventionContext context, CancellationToken cancellationToken)
   at McMaster.Extensions.CommandLineUtils.Conventions.ExecuteMethodConvention.<>c__DisplayClass0_0.<<Apply>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync(String[] args, CancellationToken cancellationToken)
   at McMaster.Extensions.CommandLineUtils.CommandLineApplication.ExecuteAsync[TApp](CommandLineContext context, CancellationToken cancellationToken)
   at NuGetUtility.Program.Main(String[] args) in /home/runner/work/nuget-license/nuget-license/src/NuGetUtility/Program.cs:line 86
   at NuGetUtility.Program.<Main>(String[] args)
Aborted (core dumped)
```

**Comment by sensslen** [comment]
OK, so the first error appears to be an issue with the frequency we're downloading files from nuget. I'll see what I can do here. For the second error, this is an issue with the nuget package. Following the link through the web nuget (https://www.nuget.org/packages/NETStandard.Library/) the license is not available. There is nothing I can do about that. This needs to be fixed by the package maintainer (Microsoft).

**Comment by trampster** [comment]
If the license is not available can you output a warning instead of throwing an exception and crashing. This way we can still use your tool for the rest of our licences.

**Comment by sensslen** [comment]
As I outlined above there are various mechanisms in place to either ignore or overwrite license information. I'm not sure a warning would be the right thing to do here. That said: feel free to create a PR that catches the issue and prints it as a validation error as something along those lines.

**Comment by sensslen** [comment]
@trampster I pushed #19 which should resolve the rate limiting issue (429 response). Can you please test and verify that this PR resolves said issue?

**Comment by sensslen** [comment]
Closing as this ticket has had no activity in some time and there ahs been a fix published (#19).
